src/java/org/apache/cassandra/tools/BootstrapMonitor.java

```diff
@@ -82,9 +83,19 @@ public void progress(String tag, ProgressEvent event)
             message = message + " (progress: " + (int)event.getProgressPercentage() + "%)";
         }
         out.println(message);
+        if (type == ProgressEventType.ERROR)
+        {
+            error = new RuntimeException(String.format("Bootstrap resume has failed with error: %s", message));
+            condition.signalAll();
```

Comment:
@maedhroz Since I reverted my changes where I removed the `ProgressEventType.COMPLETE` event that was being sent, I assume we can just rely on the logic below (lines 91-94) where it already calls `condition.signalAll();`?   When we encounter an issue, I noticed that we always both `ProgressEventType.ERROR` and `ProgressEventType.COMPLETE` events so it seems a little redundant and unnecessary now.

Reply:
I definitely wouldn't do it this way from scratch, but it's probably just not worth the compatibility headache. What we have now LGTM